ev3dev/helper.py

```diff
@@ -146,6 +147,217 @@ class MediumMotor(ev3dev.auto.MediumMotor, MotorMixin):
+        self._run_command(**kwargs)
+
+    def run_timed(self, **kwargs):
+        kwargs['command'] = ev3dev.auto.LargeMotor.COMMAND_RUN_TIMED
+        self._run_command(**kwargs)
+
+    def run_direct(self, **kwargs):
+        kwargs['command'] = ev3dev.auto.LargeMotor.COMMAND_RUN_DIRECT
+        self._run_command(**kwargs)
+
+    def reset(self, motors=None):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            motor.reset()
+
+    def stop(self, motors=None):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            motor.stop()
+
+    def _is_state(self, motors, state):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            if state not in motor.state:
+                return False
+
+        return True
+
+    @property
+    def is_running(self, motors=None):
+        return self._is_state(motors, ev3dev.auto.LargeMotor.STATE_RUNNING)
+
+    @property
+    def is_ramping(self, motors=None):
+        return self._is_state(motors, ev3dev.auto.LargeMotor.STATE_RAMPING)
+
+    @property
+    def is_holding(self, motors=None):
+        return self._is_state(motors, ev3dev.auto.LargeMotor.STATE_HOLDING)
+
+    @property
+    def is_overloaded(self, motors=None):
+        return self._is_state(motors, ev3dev.auto.LargeMotor.STATE_OVERLOADED)
+
+    @property
+    def is_stalled(self):
+        return self._is_state(motors, ev3dev.auto.LargeMotor.STATE_STALLED)
+
+    def wait(self, cond, timeout=None, motors=None):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            motor.wait(cond, timeout)
+
+    def wait_until_not_moving(self, timeout=None, motors=None):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            motor.wait_until_not_moving(timeout)
+
+    def wait_until(self, s, timeout=None, motors=None):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            motor.wait_until(s, timeout)
+
+    def wait_while(self, s, timeout=None, motors=None):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            motor.wait_while(s, timeout)
+
+
+class MotorPair(MotorSet):
+
+    def __init__(self, motor_specs, desc=None):
+        MotorSet.__init__(self, motor_specs, desc)
+        (self.left_motor, self.right_motor) = self.motors.values()
+        self.max_speed = self.left_motor.max_speed
+
+    def set_speed_ratio(self, left_motor_ratio, right_motor_ratio, set_right=True):
+        """
+        Set the speeds of the left_motor vs right_motor by ratio
+        """
+
+        # Set the right_motor speed relative to the left_motor_speed
+        if set_right:
```

Comment:
Huh, this isn't the behavior I had in mind :laughing: I was thinking you would specify a speed and ratio, and it would run the fastest motor at that speed and the other one at a speed defined by the supplied ratio. My thinking was that we'd mimic the "steering" behavior of EV3-G.

Reply:
Something like this?
https://github.com/ddemidov/ev3dev-lang-python/blob/master-deprecated/ev3dev/__init__.py#L85-L121

Reply:
I haven't touched EV3-G since the day I found ev3dev :) Does its steering do what ddemidov points to or something else?

Reply:
Looking at the "synchronized steer" code [here](https://github.com/mindboards/ev3sources/blob/master/lms2012/d_pwm/Linuxmod_AM1808/d_pwm.c#L1963), I think I can say that they _don't_ match but @ddemidov's implementation would be preferred. LEGO seems to designate one motor as a "master", drive that one at the requested speed, and then run the other at a rate defined by the steering. I am no expert though so I may be mis-reading it.